example/run.sh

```diff
@@ -136,83 +136,87 @@ echo ""
-    --label_file $out_dir/test_data.h5 --task both \
-    --test_file $out_dir/inference_latest/test_data_HSC.5M.output.h5 \
-    --sep_peaks --thresholds 0.5
-
-echo ""
-echo "Step 8: Postprocess the model output to produce bedGraph and bigWig files..."
-echo ""
-# NOTE - replace 3rd argument with your inference result file!
-# To get predicted coverage track
-python $root_dir/postprocess.py \
-    $out_dir/example.holdout_intervals.bed \
-    $out_dir/inference_latest/test_data_HSC.5M.output.h5 \
-    $ref_dir/hg19.auto.sizes \
-    $out_dir/HSC.5M.output.track \
-    --channel 0 \
-    --round 3 \
-    --num_worker -1
-# To get predicted peaks
-python $root_dir/postprocess.py \
-    $out_dir/example.holdout_intervals.bed \
-    $out_dir/inference_latest/test_data_HSC.5M.output.h5 \
-    $ref_dir/hg19.auto.sizes $out_dir/HSC.5M.output.peaks \
-    --channel 1 --threshold 0.5
+    --label_file $out_dir/test_data.h5 --task regression \
+    --test_file $out_dir/inference_latest/test_data_HSC.5M.output.track.bw \
+    --intervals $out_dir/example.holdout_intervals.bed \
+    --sizes $ref_dir/hg19.auto.sizes \
+    --sep_peaks
 
 echo ""
-echo "Step 9: Summarize the predicted peaks in a BED file..."
+echo "Step 7b: Calculate metrics for peak classification after inference..."
 echo ""
-python $root_dir/peaksummary.py \
-    --peakbw $out_dir/HSC.5M.output.peaks.bw \
-    --scorebw $out_dir/HSC.5M.output.track.bw \
-    --prefix $out_dir/HSC.5M.output.peaks.summarized
+python $root_dir/calculate_baseline_metrics.py \
+    --label_file $out_dir/test_data.h5 --task classification \
+    --test_file $out_dir/inference_latest/test_data_HSC.5M.output.peaks.bw \
+    --intervals $out_dir/example.holdout_intervals.bed \
+    --sizes $ref_dir/hg19.auto.sizes \
+    --thresholds 0.5
 
 #######
 
 echo ""
 echo "An alternative method to call peaks..."
 echo ""
 
-# To get predicted probabilities for peaks
-python $root_dir/postprocess.py \
-    $out_dir/example.holdout_intervals.bed \
-    $out_dir/inference_latest/test_data_HSC.5M.output.h5 \
-    $ref_dir/hg19.auto.sizes $out_dir/HSC.5M.output.probs \
-    --channel 1 --round 3
+python $root_dir/main.py --infer \
+    --infer_files $out_dir/test_data.h5 \
+    --intervals_file $out_dir/example.holdout_intervals.bed \
+    --sizes_file $ref_dir/hg19.auto.sizes \
+    --round 3 \
+    --weights_path $out_dir/HSC.5M.model_latest/model_best.pth.tar \
+    --out_home $out_dir --label inference \
+    --result_fname HSC.5M.output.probs \
+    --model resnet --nblocks 5 --nfilt 15 --width 50 --dil 8 \
+    --nblocks_cla 2 --nfilt_cla 15 --width_cla 50 --dil_cla 10 \
+    --task both --num_workers 0 --infer_format bigiwg
 
-macs2 bdgpeakcall -i $out_dir/HSC.5M.output.probs.bedGraph -o $out_dir/HSC.5M.output.peaks.narrowPeak -c 0.5
+macs2 bdgpeakcall -i $out_dir/inference_latest/test_data_HSC.5M.output.probs.peaks.bedGraph -o $out_dir/inference_latest/test_data_HSC.5M.output.peaks.narrowPeak -c 0.5
```

Comment:
is this `macs2` call supposed to run the output of the previous infer call? That runs seems to be outputing the data in `bigwig` format, but the macs call is taking in a `bedGraph` file?

Reply:
Regardless of the format requested, bedGraph files are generated. If --infer_format = bigiwig then bedgraph files are then converted to bigwig. Right now, since we have a need for both formats, I am not deleting the bedGraph files generated if --infer_format = bigwig. Makes sense ?

Reply:
Yep, makes sense!